### DISCUSSION TITLE
Expense/revenue reports include unwanted transactions in some circumstances

### DISCUSSION THREAD

**Discussion by canoine** [firefly-iii/firefly-iii#6854] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hello,

In my personal instance, when requesting an expense/revenue report on one or some accounts, sometimes the report also selects transactions whose description contains a question mark (?), even if the relating account wasn't chosen.

I'm not able to tell exactly what to do or not, but I think you have to select only expense accounts, not revenue accounts, to see the bug.

### Expected behaviour

Those transactions should not appear on this report.

### Debug information

Debug information generated at 2023-01-12 15:01:07 Europe/Paris for Firefly III version **5.7.18**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.18 |
| Firefly III API | 1.5.6 |
| Build | (unknown), (unknown) |
| DB version | 18 (exp. 18) |
| PHP | `8.0.25` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Paris |
| Browser TZ | Europe/Paris |
| App environment | local |
| App debug mode | false |
| App cache driver | redis |
| App logging | notice, syslog |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | fr_FR |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-01-01 00:00:00 |
| Session end | 2023-07-01 00:00:00 |
| Session first | 2020-07-31 00:00:00 |
| User ID | 2 |
| User language | fr_FR |
| User locale | fr_FR |
| Attempt at "fr_FR.utf8" | 'fr_FR.utf8' |
| Attempt at "fr_FR.UTF-8" | 'fr_FR.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:108.0) Gecko/20100101 Firefox/108.0 |

### Steps to reproduce

1/ Ensure you have at least, one revenue account, named "R" below, one expense account, named "E", and one Asset account, say "A" ; 
2/ Create a transaction "T1" from account "R" to "A", and put a "?" inside the description ;
3/ Create a transaction "T2" from account "A" to "E" ;
4/ Go to the report page, and select  :
- Expense/revenue report
- Account "A"
- The appropriate dates
- Account "E", but not "R"

5/ See that "T1" and "E" appear inside the report.

### Additional info

The first line of the "Expense / gain by account" table (top-left) shows  "R" / "E".

**Comment by JC5** [comment]
Mmm, thanks. I'll check it out.

**Comment by JC5** [comment]
I can't replicate this.

https://docs.firefly-iii.org/firefly-iii/advanced-concepts/reports/#expenserevenue-report

**Comment by canoine** [comment]
OK. May I do something to help ?

**Reply by JC5** [reply]
In your example, the account "E" would not have been an option to select in the expense/revenue report. Trying to replicate the issue that unwanted transactions are visible in the report fail. I'm curious to learn what you tried to do exactly.

**Comment by canoine** [comment]
The problem is... I don't know.  I'm trying to reproduce the problem by adding new "buggy" operations, but no success for now. I don't even know which accounts appear and doesn't appear into the report accounts list and why. Even more, sometimes this list grows when I try to change the report parameters... So I think my database is corrupted. I'll try to restore the data from a dump, to see my problem persists.

**Comment by canoine** [comment, marked as answer]
So... The problem definitely isn't the question mark. It was (yes, I say "was") the revenue account "R". After several tries (new operations, clones, with one and only one difference, never the same, you know), all the operations from R, and only from R appear.
Until I rename this account :
- "R" > "test" : OK, all is well ;
- then "test" >"R" : OK too. The bug automagically disappeared.

Good, but not good, as I don't know what happened, and I can't reproduce no more. I guess the database entry was corrupted, and updating it "decorrupted" it. I hate not to know.

So, sorry for the noise, and thank you for trying to solve my problem.
